src/ai/backend/common/utils.py

```diff
@@ -478,3 +478,15 @@ def deep_merge(*args: Mapping[str, Any]) -> Mapping[str, Any]:
             else:
                 merged[k] = vb
     return merged
+
+
+def addr_to_hostport_pair(addr: str) -> tuple[str, int]:
```

Comment:
If there’s already an existing utility function, it would be better to use that instead.